experiments/modelfree/plot_noisy_observations.py

```diff
@@ -0,0 +1,141 @@
+import json
+import math
+import os
+
+from matplotlib import pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+RESULT_DIR = "/Users/cody/Data/adversarial_policies/ray_results/noisy_victim_obs/"
+OUT_DIR = "/Users/cody/Data/adversarial_policies/noisy_obs_plots/results_may_22"
+env_lookup = {
+    'SumoHumans': 'multicomp/SumoHumansAutoContact-v0',
+    'SumoAnts': 'multicomp/SumoAntsAutoContact-v0',
+    'KickAndDefend': 'multicomp/KickAndDefend-v0',
+    # 'YouShallNotPass': 'multicomp/YouShallNotPassHumans-v0' (Removed for now because I don't have
+    # time at the moment to fix the plotting logic to deal with victim in different location)
+}
+available_zoos = {
+    'SumoHumans': 3,
+    'SumoAnts': 4,
+    'KickAndDefend': 3
+}
+
+k_keys = ['env', 'agent0_type', 'agent0_path', 'agent1_type',
+          'agent1_path', 'masking_param', 'noise_magnitude']
+
+
+def transform(df, transform_list):
+    new_df = df.copy()
+    for trans_dict in transform_list:
+        new_df[trans_dict['new_col']] = new_df[trans_dict['old_col']].apply(trans_dict['func'])
+    return new_df
+
+
+def subset(df, spec):
+    ret = df.copy()
+    for constraint, constraint_value in spec.items():
+        ret = ret[ret[constraint] == constraint_value]
+    return ret
+
+
+def process_element_into_flat_dict(el, key_order=k_keys):
+    outp = {}
+    for i, k in enumerate(key_order):
+        outp[k] = el['k'][i]
+    outp['agent0_wins'] = el['v']['win0']
+    outp['agent1_wins'] = el['v']['win1']
+    outp['ties'] = el['v']['ties']
+    return outp
+
+
+def noisy_adversary_opponent_subset_plot(original_df, subset_specs, transform_specs,
+                                         logistic=True, plot_line=True, savefile=None):
+    subset_df = subset(original_df, subset_specs)
+    if len(subset_df) == 0:
+        return
+    transformed_df = transform(subset_df, transform_specs)
+    plt.figure(figsize=(10, 7))
```

Comment:
Nit: using `plt` is error-prone as it defaults to the latest figure. Good practice is to use `fig = plt.figure`, and then the corresponding `Figure` instance methods: `fig.suptitle`, `fig.savefig`, `fig.show`, `fig.close`. (You don't need to change this here but I'd recommend it for new code, especially if you have any complicated nested plotting.)